lib/foreman_discovery/engine.rb

```diff
@@ -15,6 +15,8 @@ class Engine < ::Rails::Engine
     config.autoload_paths += Dir["#{config.root}/app/models/concerns"]
     config.autoload_paths += Dir["#{config.root}/app/services"]
 
+    config.eager_load_paths += Dir["#{config.root}/lib"]
```

Comment:
This feels very weird. This file itself is in `lib` so isn't it already loaded at this point?

Reply:
I am not very familiar with how foreman loads the plugins, maybe it is looking for a specific engine file? And the rest of the lib is ignored?

I will try tomorrow the solution you suggested in nightly, meanwhile adding @ShimShtein to the party :)